src/module/os.c

```diff
@@ -81,9 +81,18 @@ void processCwd(WrenVM* vm)
 {
   wrenEnsureSlots(vm, 1);
 
-  char buffer[WREN_PATH_MAX * 4];
-  size_t length = sizeof(buffer);
-  if (uv_cwd(buffer, &length) != 0)
+  char _buffer[WREN_PATH_MAX * 2 + 1];
+  char* buffer = _buffer;
+  size_t length = sizeof(_buffer);
+  int result = uv_cwd(buffer, &length);
+
+  if (result == UV_ENOBUFS)
+  {
+    buffer = (char*)malloc(length);
```

Comment:
Now we need something like `if (buffer != _buffer) free(buffer);` at the end.

Reply:
Added. 👍